src/app/api/services/dot-menu.service.ts

```diff
@@ -73,6 +73,17 @@ export class DotMenuService {
         return this.loadMenu();
     }
 
+    /**
+     * Return the DotMenu's ID of a Portlet
+     * @param portletId MenuItems id
+     */
+    getDotMenuId(portletId: string): Observable<string> {
+        return this.loadMenu()
+            .flatMap((menus: DotMenu[]) => menus)
+            .find((menu: DotMenu) => menu.menuItems.some(menuItem => menuItem.id === portletId))
+            .map((menu: DotMenu) => menu.id);
```

Comment:
I think you can do a second ```.flatMap((menu: DotMenu) => menu.menuItems)``` (or a map) and that way you don't need to use a ```.some```

Reply:
If I did it on this way I would get MenuItem for the last map like this:

.map((menuItem: MenuItem) => ...);

... and I need return the DotMenu's id.